Memorize/MemoryGame.swift

```diff
@@ -8,79 +8,81 @@
 import Foundation
 
 struct MemoryGame<CardContent> where CardContent: Equatable {
-
     // MARK: - Properties
+    
     struct Card: Identifiable {
-
         var isFaceUp: Bool = false
         var isMatched: Bool = false
         var content: CardContent
         var isPreviouslySeen: Bool = false
         var id: Int
-
     }
-
     private(set) var cards: [Card]
-//    private var indexOfTheOneAndOnlyFaceUpCard: Int?
     private var indexOfTheOneAndOnlyFaceUpCard: Card?
     private(set) var score: Int = 0
-
+    
     // MARK: - Initilization
+    
     init(numberOfPairsOfCards: Int, createCardContent: (Int) -> CardContent) {
         cards = [Card]()
         for pairIndex in 0..<numberOfPairsOfCards {
             let content: CardContent = createCardContent(pairIndex)
             cards.append(Card(content: content, id: pairIndex*2))
-            cards.append(Card(content: content, id: pairIndex * 2 + 1))
+            cards.append(Card(content: content, id: pairIndex*2 + 1))
         }
         cards.shuffle()
     }
-
+    
     // MARK: - Functions
+    
     mutating func choose(_ card: Card) {
+        """
+        This function is used to update the score and card details to statisy the following game conditions:
+        1. At a time only two cards should be faceUp
+        2. If the content of the card is matching then update the score by 2
+        3. else facedown all the cards and if the if the any one of the card is already seen then reduce score by 1
+        """
+        
         guard
             var chosenCard = cards.first(where: {$0.id == card.id}),
             !chosenCard.isFaceUp,
             !chosenCard.isMatched
-        else {
-            return
-        }
-        print("Chosen Card Initially \(chosenCard)")
+        else { return }
+        
         if var potentiallyMatchCard =  indexOfTheOneAndOnlyFaceUpCard {
             if chosenCard.content == potentiallyMatchCard.content {
                 chosenCard.isMatched = true
                 potentiallyMatchCard.isMatched = true
-                score += 2
-            } else {
-                // If not matched and card is already seen then score should reduce by 1
-                if chosenCard.isPreviouslySeen ||
-                   potentiallyMatchCard.isPreviouslySeen {
-                    score -= 1
-                }
+                score += 2  // If cards contain is matched then increase the score by 2
+                updateOriginalCard(chosenCard)
+                updateOriginalCard(potentiallyMatchCard)
```

Comment:
Is this function updating the card you are passing in as an argument?  Or is this function updating the "original card" (whatever "original card" means)?  If it's only updating the card you are passing in, then would a clearer call site be `update(choseCard)` or maybe `updateDeck(with: chosenCard)`?